lib/HLSL/DxilValidation.cpp

```diff
@@ -2134,17 +2134,35 @@ static void ValidateBarrierFlagArg(ValidationContext &ValCtx, CallInst *CI,
   }
 }
 
+std::string GetLaunchTypeStr(DXIL::NodeLaunchType LT) {
+  switch (LT) {
+  case DXIL::NodeLaunchType::Broadcasting:
+    return "Broadcasting";
+  case DXIL::NodeLaunchType::Coalescing:
+    return "Coalescing";
+  case DXIL::NodeLaunchType::Thread:
+    return "Thread";
+  case DXIL::NodeLaunchType::Invalid:
+  case DXIL::NodeLaunchType::LastEntry:
+    return "Invalid";
+  }
+  llvm_unreachable("invalid launch type");
```

Comment:
I think within the context of the validator, we should not use llvm_unreachable here, since you certainly could have an invalid value that you're passing to validation.  Instead, we should probably have Invalid/LastEntry/default case that simply returns "<invalid>".  That is unless we check for invalid launch type elsewhere first, in which case all of these should be unreachable.

Reply:
FYI: "" was mis-rendered.  It was written as `"<invalid>"`, and the middle was dropped when rendering the comment!